src/EditorFeatures/CSharpTest/MisplacedUsingDirectives/AbstractMisplacedUsingDirectivesCodeFixProviderTests.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.AddImports;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.MisplacedUsingDirectives;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Testing;
+using Microsoft.CodeAnalysis.Testing.Verifiers;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.MisplacedUsingDirectives
+{
+    /// <summary>
+    /// Base test class for the <see cref="MisplacedUsingDirectivesCodeFixProvider"/>.
+    /// </summary>
+    public abstract class AbstractMisplacedUsingDirectivesCodeFixProviderTests
```

Comment:
this seems strange that it's not subclassing our normal IDE code fix test harness.